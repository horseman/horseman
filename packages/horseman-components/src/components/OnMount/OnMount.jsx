import React from "react";
import PropTypes from "prop-types";

class OnMount extends React.Component {
  componentDidMount() {
    const { gtmEvent } = this.props;

    if (gtmEvent) {
      this.fireGtmEvent();
    }
  }

  fireGtmEvent() {
    const { gtmEvent } = this.props;
    if (
      typeof window !== "undefined" &&
      typeof window.dataLayer !== "undefined"
    ) {
      window.dataLayer.push({ event: gtmEvent });
    }
  }

  render() {
    return <div>{this.props.children}</div>;
  }
}

OnMount.propTypes = {
  gtmEvent: PropTypes.string,
  children: PropTypes.node,
};

export default OnMount;
