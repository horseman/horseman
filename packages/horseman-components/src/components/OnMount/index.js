import OnMount from "./OnMount";

export default OnMount;
