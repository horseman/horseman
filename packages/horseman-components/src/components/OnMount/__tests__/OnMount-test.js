import React from "react";
import { shallow } from "enzyme";

import OnMount from "../";

const TestComponent = () => <div />;

describe("OnMount", () => {
  test("Fires Component Did Mount only", () => {
    const spy = jest.spyOn(OnMount.prototype, "componentDidMount");
    const wrapper = shallow(
      <OnMount>
        <TestComponent />
      </OnMount>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
  });

  test("Fires fireGtmEvent", () => {
    window.dataLayer = {
      push: obj => obj,
    };
    const windowSpy = jest.spyOn(window.dataLayer, "push");

    const spy = jest.spyOn(OnMount.prototype, "fireGtmEvent");
    const wrapper = shallow(
      <OnMount gtmEvent="loaded">
        <TestComponent />
      </OnMount>,
    );
    expect(wrapper).toMatchSnapshot();
    const event = "loaded";
    expect(spy).toHaveBeenCalled();
    expect(windowSpy).toHaveBeenCalledWith({ event });
  });
});
