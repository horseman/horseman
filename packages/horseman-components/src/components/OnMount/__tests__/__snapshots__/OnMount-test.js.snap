// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnMount Fires Component Did Mount only 1`] = `
<div>
  <TestComponent />
</div>
`;

exports[`OnMount Fires fireGtmEvent 1`] = `
<div>
  <TestComponent />
</div>
`;
